Source/WebCore/html/CanvasBase.cpp

```diff
@@ -334,18 +334,16 @@ RefPtr<ImageBuffer> CanvasBase::allocateImageBuffer() const
         return nullptr;
     }
 
+    auto* context = renderingContext();
+    auto colorSpace = context ? context->colorSpace() : DestinationColorSpace::SRGB();
+    auto pixelFormat = context ? context->pixelFormat() : PixelFormat::BGRA8;
+    auto willReadFrequently = context ? context->willReadFrequently() : false;
+
     OptionSet<ImageBufferOptions> bufferOptions;
-    if (shouldAccelerate(area))
+    if (!willReadFrequently && shouldAccelerate(area))
         bufferOptions.add(ImageBufferOptions::Accelerated);
-
-    auto colorSpace = DestinationColorSpace::SRGB();
-    auto pixelFormat = PixelFormat::BGRA8;
-
-    if (auto* context = renderingContext()) {
+    if (context)
         bufferOptions = context->adjustImageBufferOptionsForTesting(bufferOptions);
-        colorSpace = context->colorSpace();
-        pixelFormat = context->pixelFormat();
```

Comment:
E.g. this hunk could be preserved and
```
#if !PLATFORM(COCOA)
        if (context->willReadFrequently())
              bufferOptions.remove(..Accelerated..);
#endif
```

Reply:
Ok, but I took a different approach that I find easier to follow.